lib/create-branch.js

```diff
@@ -106,7 +106,32 @@ module.exports = async (
 
       const isNpm = lockfilePath.includes('package-lock.json')
       try {
-        const { ok, contents } = await getNewLockfile(commit.content, oldLockfileContent, isNpm)
+        const { tokens, 'token-audits': tokenAudits } = await dbs() // eslint-disable-line
+
+        /*
+        This is the structure of the tokens 'model'
+        _id: `${installationId}
+          tokens: {
+            ${repoId}: {
+            npm: ${token},
+            github: ${token}
+          }
+        }
+        */
+        let repositoryTokens = ''
+        try {
+          repositoryTokens = await tokens.get(installationId)
+
+          const datetime = new Date().toISOString().substr(0, 19).replace(/[^0-9]/g, '')
+
+          // write audit log entry to 'tokens-audit' db
+          // log entry type: 'read'
+          await tokenAudits.put({
+            _id: `${installationId}:${repoDoc._id}:${datetime}:read`,
+            keys: Object.keys(repositoryTokens.tokens[repoDoc._id])
+          })
+        } catch (e) {}
+        const { ok, contents } = await getNewLockfile({ packageJson: commit.content, lock: oldLockfileContent, isNpm, repositoryTokens: JSON.stringify(repositoryTokens) })
```

Comment:
repositoryTokens should also be `repositoryTokens.tokens[repoDoc._id]` so we only get the tokens we need

Reply:
+1